docs/architecture-description.md

```diff
@@ -719,20 +719,54 @@ l1RollingHashes(
 
 # Gas price setting
 
-The gas price for transaction execution in Linea is computed from L1 gas price.
-
-
-![gas price API](assets/gasPrice.drawio.svgo.svg)
-
-
-Gas price is based on the history of the gas price on L1.
-
-It uses the geth method `miner_setGasPrice` and the besu method
-
-`miner_setMinGasPrice` to update the gas price of the nodes.
-
-Both methods expect a single parameter: `0x${gasPrice.toString(16)}`
-
+There are 3 aspects of gas pricing on Linea:
+* Ensure Sequencer's inclusion logic is aligned to L1 fee market. This is to avoid exploiting Linea to execute
+transactions for unsustainably low fees
+* Ensure the best user experience for users who use Linea nodes with extended capabilities. Unlike vanilla Ethereum
+protocol, gas price on Linea and other rollups is not 2-dimensional (base fee, priority fee). There's at least L1 fees
+(execution fees and blob fees), infrastructural cost (mostly proving, but not only), potential priority fee
+(only when there's a high congestion and there's competition for L2 block space). This is an issue for interoperability,
+because vanilla Ethreum API isn't tailored for this. That's why there's a Besu plugin addressing this issue and
+providing gas price depending on input transaction
+* Ensure an OK user experience for users who run vanilla nodes. Namely, eth_gasPrice should return fees high enough, so
+99.9% of transactions are includable on Linea.
+
+This is how these challenges were solved technically:
+
+![gas price API](assets/gasPrice.drawio.svg)
+
+Coordinator fetches L1 fees data, based on which it will compute gas pricing components. There are 3:
```

Comment:
There are 3 aspects to this:

Reply:
I'll change to "there are 3 of them", I prefer components here